src/sst/elements/mercury/components/operating_system.h

```diff
@@ -16,250 +16,62 @@
+  SST_ELI_REGISTER_SUBCOMPONENT(SST::Hg::OperatingSystem, "hg",
+                                "OperatingSystem",
+                                SST_ELI_ELEMENT_VERSION(0, 0, 1),
+                                "Mercury Operating System",
+                                SST::Hg::OperatingSystemAPI)
 
-  SST_ELI_REGISTER_SUBCOMPONENT(
-    SST::Hg::OperatingSystem,
-    "hg",
-    "OperatingSystem",
-    SST_ELI_ELEMENT_VERSION(0,0,1),
-    "Mercury Operating System",
-    SST::Hg::OperatingSystem
-  )
+  OperatingSystem(SST::ComponentId_t id, SST::Params &params);
 
-  OperatingSystem(SST::ComponentId_t id, SST::Params& params, NodeBase* parent);
+  ~OperatingSystem() {}
 
-  ~OperatingSystem();
+  void init(unsigned phase) override {
+    OperatingSystemImpl::init(phase);
+  }
 
   void setup() override;
 
-  void handleEvent(SST::Event *ev) override;
-
-  bool clockTic(SST::Cycle_t) override {
-    //noop
-    return true;
-  }
-
   static size_t stacksize() {
     return sst_hg_global_stacksize;
   }
 
-  static void requireDependencies(SST::Params& params, SST::Hg::OperatingSystem& me);
-
-  std::function<void(NetworkMessage*)> nicDataIoctl() override;
-
-  std::function<void(NetworkMessage*)> nicCtrlIoctl() override;
-
-  /**
-   * @brief block Block the currently running thread context.
-   * This must be called from an application thread, NOT the DES thread
-   * @param req [in-out] A key that will store blocking data that will be needed
-   *                     for later unblocking
-   * @return
-   */
-  void block() override;
-
-  void unblock(Thread* thr) override;
-
-
-  void blockTimeout(TimeDelta delay) override;
-  // void sleep(TimeDelta time);
-  // void compute(TimeDelta time);
-
-  void startApp(App* theapp, const std::string&  /*unique_name*/) override;
-  void startThread(Thread* t) override;
-  void joinThread(Thread* t) override;
-  void scheduleThreadDeletion(Thread* thr) override;
-  void switchToThread(Thread* tothread) override;
-  void completeActiveThread() override;
-
-  static Thread* currentThread(){
-    return staticOsThreadContext()->activeThread();
-  }
-
-  static inline OperatingSystem*& staticOsThreadContext(){
-//  #if SST_HG_USE_MULTITHREAD
-    int thr = ThreadInfo::currentPhysicalThreadId();
-    return active_os_[thr];
-//  #else
-//    return active_os_;
-//  #endif
-  }
-
-  inline OperatingSystem*& activeOs() {
-//#if SST_HG_USE_MULTITHREAD
-  return active_os_[threadId()];
-//#else
-//  return active_os_;
-//#endif
-  }
-
-  Thread* activeThread() const override {
-    return active_thread_;
-  }
-
-  static OperatingSystem* currentOs(){
-    return staticOsThreadContext();
-  }
-
-  UniqueEventId allocateUniqueId() override {
-    next_outgoing_id_.msg_num++;
-    return next_outgoing_id_;
-  }
+  // Forward calls to OperatingSystemImpl here
+  #include <mercury/components/operating_system_call_forward.h>
```

Comment:
A little strange to include the same in different class declarations (the other is in operating_system_cl.h#L62) but the compiler is happy with it.